ResearchKit/Common/ORKFormItemCell.m

```diff
@@ -1021,28 +1037,22 @@ - (void)answerDidChange {
     id<ORKScaleAnswerFormatProvider> formatProvider = self.formatProvider;
     id answer = self.answer;
     if (answer && answer != ORKNullAnswerValue()) {
-        if (![self.answer isKindOfClass:[NSNumber class]]) {
-            @throw [NSException exceptionWithName:NSGenericException reason:@"Answer should be NSNumber" userInfo:nil];
-        }
         
-        [_sliderView setCurrentValue:answer];
+        [_sliderView setCurrentAnswerValue:answer];
+
     } else {
         if (answer == nil && [formatProvider defaultNumber]) {
```

Comment:
I wonder if we should rename `defaultNumber` to `defaultAnswer` on all three scale providers, and move type of answer logic to the format provider such as it returns an `NSNumber` for the classic scales, and a `NSArray<id<NSCopying, NSCoding, NSObject>>` for the text scale.

That way, the logic down here would simplify, and we'll always use `[_sliderView  setCurrentAnswerValue:...]`, which in turn internally calls `setCurrentTextChoiceValue` or `setNumberValue`. Encapsulation would feel better, I think.